File: bindings/lua/Polycode/BasicFile.lua
class "BasicFile"


function BasicFile:__getvar(name)
	if name == "file" then
		local retVal = Polycode.BasicFile_get_file(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["FILE"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function BasicFile:__setvar(name,value)
	if name == "file" then
		Polycode.BasicFile_set_file(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function BasicFile:read(ptr, size, count)
	local retVal = Polycode.BasicFile_read(self.__ptr, ptr.__ptr, size.__ptr, count.__ptr)
	return retVal
end

function BasicFile:write(ptr, size, count)
	local retVal = Polycode.BasicFile_write(self.__ptr, ptr.__ptr, size.__ptr, count.__ptr)
	return retVal
end

function BasicFile:seek(offset, origin)
	local retVal = Polycode.BasicFile_seek(self.__ptr, offset, origin)
	return retVal
end

function BasicFile:tell()
	local retVal =  Polycode.BasicFile_tell(self.__ptr)
	return retVal
end

function BasicFile:__delete()
	if self then Polycode.delete_BasicFile(self.__ptr) end
end

File: bindings/javascript/Polycode/BasicFile.js
function BasicFile() {
}

BasicFile.prototype.read = function(ptr,size,count) {
	Polycode.BasicFile_read(this.__ptr, ptr,size,count)
}

BasicFile.prototype.write = function(ptr,size,count) {
	Polycode.BasicFile_write(this.__ptr, ptr,size,count)
}

BasicFile.prototype.seek = function(offset,origin) {
	Polycode.BasicFile_seek(this.__ptr, offset,origin)
}

BasicFile.prototype.tell = function() {
	Polycode.BasicFile_tell(this.__ptr)
}
